### DISCUSSION TITLE
Issues in creating a sprite sheet/EzGif code questions

### DISCUSSION THREAD

**Discussion by lynxmeninx** [ImageMagick/ImageMagick#5339] [Help] [answered]
Hello again! - I felt this was slightly unrelated to my last question, so I wanted to post another, but if it would be better for me to combine them I totally understand and please let me know ♥ I'm using ImageMagick 7.1.0 Q16 (windows x64).

I have been trying to create several sprite sheets (each consisting of the frames from multiple different gifs). As before, the code I've been using works well, but I have a couple of problems:
```
magick convert ^
   ( character_standing.gif -coalesce +append ) ^
   ( character_walking.gif -coalesce +append ) ^
   -append -filter point -resize 50% -extent 720x720 -gravity northwest character_spritesheet.png 
```

1. I'd like to duplicate some of the frames in one of the gifs to account for the duration of each frame being different in the original gif, as this doesn't translate to a spritesheet. Say I wanted the very first frame in the sequence to be duplicated several times before the second frame appears - can I do that in a single command? (At this stage I've been opening the gif in ezgif, manually copying the frame several times, and then saving it, and _then_ running it through the above command; but doing that for every single character will be quite tiresome!)

2. [Moved to another discussion!]

I know I'm asking a heck of a lot here, but I've been scouring the documentation and (at least with my limited understanding of anything code related) I haven't been able to find an answer to either of these questions. If you know how I could go about these, I'd really appreciate hearing from you!!

Thank you so much again :)

**Comment by fmw42** [comment]
Can you post an example set of input images to help us understand your issue and to work on

**Reply by lynxmeninx** [reply]
Will do - at the risk of being a total nuisance, I'm going to create another discussion just for my -extent issue. Thank you :D

**Comment by snibgo** [comment, marked as answer]
I sugest using just "magick" instead of "magick convert".

1. After you have coalesced frames, before appending them, you can can manipulate the image list. In your case, suppose you want to duplicate the first frame (image index 0) 4 times. `duplicate 4,0` does that. The new images will be at the end of the list, but you want them at the start. One way of doing that is to reverse the list before and after the operation. Then we need to duplicate the *last* image, index -1.
```
 -reverse -duplicate 4,-1 -reverse
```

2. You don't show your command, or input or output. Commenting is difficult.

**Reply by lynxmeninx** [reply]
That's awesome!! Thank you so much for the answer!
On 2., I'm going to make another discussion to show my issue, as I'm having a bit of trouble wording it concisely.
